library/modules/Maps.cpp

```diff
@@ -713,6 +715,50 @@ bool Maps::canStepBetween(df::coord pos1, df::coord pos2)
     return false;
 }
 
+/*
+* Plants
+*/
+df::plant *Maps::getPlantAtCoords(int32_t x, int32_t y, int32_t z)
+{
+    if (x < 0 || x >= world->map.x_count || y < 0 || y >= world->map.y_count || !world->map.column_index)
+        return NULL;
+
+    df::map_block_column *mbc = world->map.column_index[(x / 48) * 3][(y / 48) * 3];
+    if (!mbc)
+        return NULL;
+
+    int32_t x_mod_48 = x % 48;
+    int32_t y_mod_48 = y % 48;
```

Comment:
Another more minor thing: would a more descriptive name like `region_x` work here? (Trying to use terminology from https://docs.dfhack.org/en/stable/docs/api/Maps.html, although I would still want to make it clear that these are coordinates _within_ a region, not _of_ a region).

These are only used once below, and I personally don't find `x_mod_48` more clear than just `(x % 48)`, so I'd mainly just suggest keeping this for readability. (If calculated with `x % 48` in the loop below, I'd expect most basic optimizations to result in the value just being calculated once anyway, so hand-optimizing may not be necessary here.)